rosetta/transactions/parse_transaction.go

```diff
@@ -16,15 +16,107 @@ package transactions
 
 import (
 	"fmt"
+	"strconv"
 
+	"github.com/onflow/cadence/encoding/json"
 	"github.com/onflow/flow-go-sdk"
 
+	"github.com/optakt/flow-dps/models/dps"
 	"github.com/optakt/flow-dps/rosetta/identifier"
 	"github.com/optakt/flow-dps/rosetta/object"
 )
 
+const (
+	// Transactions should have exactly one authorizer.
+	authorizersRequired = 1
+
+	// Transaction script should have exactly two arguments.
+	argsRequired = 2
+)
+
 // ParseTransactions processes the flow transaction and translates it to a list of operations and a list of
 // signers.
 func (p *Parser) ParseTransaction(tx *flow.Transaction) ([]object.Operation, []identifier.Account, error) {
-	return nil, nil, fmt.Errorf("TBD: not implemented")
+
+	// Verify that we have the correct number of authorizers.
+	if len(tx.Authorizers) != authorizersRequired {
+		return nil, nil, fmt.Errorf("invalid authorizer count (have: %v, want: %v)", len(tx.Authorizers), authorizersRequired)
+	}
+
+	// Verify that the transaction script has the correct number of arguments.
+	args := tx.Arguments
+	if len(args) != argsRequired {
+		return nil, nil, fmt.Errorf("invalid arguments count: (have: %v, want: %v)", len(args), argsRequired)
+	}
+
+	// Parse the amount script argument.
+	val, err := json.Decode(args[0])
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not parse transaction amount: %w", err)
+	}
+	amountArg, ok := val.ToGoValue().(uint64)
+	if !ok {
+		return nil, nil, fmt.Errorf("invalid transaction amount: %v", val.String())
+	}
+	amount := strconv.FormatUint(amountArg, 10)
+
+	// Parse the receiver script argument.
+	val, err = json.Decode(args[1])
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not parse receiver address: %w", err)
+	}
+	receiver := flow.HexToAddress(val.String())
+
+	ops := make([]object.Operation, 2)
+
+	// Create the send operation.
+	ops[0] = object.Operation{
+		ID: identifier.Operation{
+			Index: 0,
+		},
+		RelatedIDs: []identifier.Operation{{Index: 1}},
+		AccountID: identifier.Account{
+			Address: tx.Authorizers[0].String(),
+		},
+		Type:   dps.OperationTransfer,
+		Status: dps.StatusCompleted,
```

Comment:
This we don't validate on the input to the `parse` endpoint, but I've filled it here. Could also remove it so the actions are more symmetrical.

Reply:
Automatically filling anything here undermines the purpose of this endpoint to serve as a security fallback / sanity check. Every information should be derived from the transaction. I think you might have to take a bit of a step back and invest more time into this endpoint.

In this case, we should probably match the transaction script against a regular expression for the transfer script to verify it's a transfer and to extract relevant information such as token name. It's basically about doing the reverse of what the script generator does when we create the transaction. Feel free to put a `Parser` or `Interpreter` into the scripts folder who can do that, for example.

Also, as far as I'm aware, the status should be part of the transaction, so we should have a mapping between Flow status and Rosetta status.